From 9429d621bb2815160a5c42e0205921d4b977c7d0 Mon Sep 17 00:00:00 2001
From: OpenEmbedded <oe.patch@oe>
Date: Mon, 29 Apr 2024 14:18:26 +0000
Subject: [PATCH] kernel: workqueue: set kworker to sched_fifo

Make sure that the kworker threads are running with the SCHED_FIFO.
---
 kernel/workqueue.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kernel/workqueue.c b/kernel/workqueue.c
index f5fa7be8d17e..ee6bc25ee083 100644
--- a/kernel/workqueue.c
+++ b/kernel/workqueue.c
@@ -1969,6 +1969,7 @@ static struct worker *create_worker(struct worker_pool *pool)
 		goto fail;
 
 	set_user_nice(worker->task, pool->attrs->nice);
+	sched_set_fifo(worker->task);
 	kthread_bind_mask(worker->task, pool->attrs->cpumask);
 
 	/* successful, attach the worker to the pool */
-- 
2.40.1

